### PR TITLE
Fix some filtering bugs

### LINKED ISSUE #17: Filter function
Am I missing something or is the filter function really case sensitive and substrings are not found?

filter_link_editor

### LINKED ISSUE #19: "Link all" with filtered features
If I filter features on the left side, I would expect to only link the filtered features when I choose "link all"

image

Now all features are linked with "link all", which is a bit unexpected and not really userfriendly in my eyes...

image

### PR DESCRIPTION
**Pull request by domi4484**
- Fix #17 Filter should be case insensitiv
- Fix #19 Link all button should take care of filter (was the case only for quick filter)

### CODE REVIEW

File: linking_relation_editor/core/features_model_filter.py [hunk 1]
```diff
@@ -137,12 +121,11 @@ def filterAcceptsRow(self,
...
             return True
 
-        rowDisplayRole = self.sourceModel().data(index,
-                                                 Qt.DisplayRole)
+        rowDisplayRole = self.sourceModel().data(index, Qt.DisplayRole)
         if not rowDisplayRole:
             return False
 
-        return self._quick_filter in rowDisplayRole
+        return self._quick_filter.lower() in rowDisplayRole.lower()
```
Comment (signedav) [inline linking_relation_editor/core/features_model_filter.py:132]: Besides all the formatting diffs this is the only change here?

**Review by signedav — APPROVED** [review]

File: linking_relation_editor/core/features_model_filter.py [hunk 2]
```diff
@@ -56,10 +47,11 @@ def clear_quick_filter(self):
         self.invalidateFilter()
 
     def quick_filter_active(self):
-        return len(self._quick_filter) > 0
+        return (
+            self._quick_filter or self._map_filter or self._feature_filter != FeaturesModelFilter.FeatureFilter.ShowAll
+        )
```
Comment (domi4484) [inline linking_relation_editor/core/features_model_filter.py:52]: @signedav this is the second fix

File: linking_relation_editor/core/features_model_filter.py (hunk 1 above)
Comment (domi4484) [reply linking_relation_editor/core/features_model_filter.py:132]: For the first fix yes

File: linking_relation_editor/core/features_model_filter.py (hunk 2 above)
Comment (signedav) [reply linking_relation_editor/core/features_model_filter.py:52]: Cool thanks

**Pull request merged**